src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+// write code here;
+
+export function groupByKey<T>(arrayOfObjects: T[], key: keyof T): GroupsMap<T> {
+  const newArrayOfObjects: GroupsMap<T> = {};
+
+  arrayOfObjects.map((objectInArray) => {
+    const keyValue = String(objectInArray[key]);
```

Comment:
Casting the `key` of `objectInArray` to a string is a good approach to ensure that the keys of the `GroupsMap` are strings. However, it's important to note that this will convert all key values to their string representation, which may not be desired for all data types (e.g., objects, symbols). If the function is intended to handle only primitive types as keys, this should be clearly documented.